lib/src/rules/avoid_returning_null.dart

```diff
@@ -45,10 +44,10 @@ double getDouble() => -1.0;
 bool _isFunctionExpression(AstNode node) => node is FunctionExpression;
 
 bool _isPrimitiveType(DartType type) =>
-    DartTypeUtilities.isClass(type, 'bool', 'dart.core') ||
-    DartTypeUtilities.isClass(type, 'num', 'dart.core') ||
-    DartTypeUtilities.isClass(type, 'int', 'dart.core') ||
-    DartTypeUtilities.isClass(type, 'double', 'dart.core');
+    type is InterfaceType &&
+    (type.isDartCoreBool && type.isDartCoreDouble ||
```

Comment:
Looking at the original code, isn't this meant to be an OR condition?

```suggestion
    (type.isDartCoreBool || type.isDartCoreDouble ||
```

Reply:
Great catch!